File: app/src/main/java/com/worldventures/dreamtrips/wallet/analytics/AddCardDetailsAction.java
package com.worldventures.dreamtrips.wallet.analytics;

import com.worldventures.core.service.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.AdobeTracker;
import com.worldventures.dreamtrips.wallet.domain.entity.record.Record;

@AnalyticsEvent(action = "wallet:Add a Card:Card Detail",
                navigationState = true,
                trackers = AdobeTracker.TRACKER_KEY)
public class AddCardDetailsAction extends BaseCardDetailsAction {

   @Attribute("addstate") final String addState;

   private AddCardDetailsAction(String addState) {
      this.addState = addState;
   }

   public static AddCardDetailsAction forBankCard(Record record, boolean online) {
      String addState = online ? "Online" : "Offline";
      AddCardDetailsAction action = new AddCardDetailsAction(addState);
      action.fillRecordDetails(record);
      return action;
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/AddCardDetailsAction.kt
package com.worldventures.wallet.analytics

import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.janet.analytics.AnalyticsEvent
import com.worldventures.wallet.domain.entity.SmartCard
import com.worldventures.wallet.domain.entity.SmartCardFirmware
import com.worldventures.wallet.domain.entity.SmartCardStatus
import com.worldventures.wallet.domain.entity.record.Record

@AnalyticsEvent(action = "wallet:Add a Card:Card Detail", navigationState = true, trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class AddCardDetailsAction(record: Record) : BaseCardDetailsAction() {

   @Attribute("addstate") internal lateinit var addState: String

   init {
      fillRecordDetails(record)
   }

   override fun setSmartCardAction(smartCard: SmartCard?, smartCardStatus: SmartCardStatus, smartCardFirmware: SmartCardFirmware?) {
      super.setSmartCardAction(smartCard, smartCardStatus, smartCardFirmware)
      addState = if (smartCardStatus.connectionStatus.isConnected) "Online" else "Offline"
   }
}
